core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -208,30 +237,78 @@ object ConsumerGroupCommand {
       case Some(brokerId) =>
-        val consumerOpt = getConsumer(zkUtils, brokerId)
-        consumerOpt match {
-          case Some(consumer) =>
-            val request =
-              OffsetRequest(Map(topicAndPartition -> PartitionOffsetRequestInfo(OffsetRequest.LatestTime, 1)))
-            val logEndOffset = consumer.getOffsetsBefore(request).partitionErrorAndOffsets(topicAndPartition).offsets.head
-            consumer.close()
-
-            val lag = offsetOpt.filter(_ != -1).map(logEndOffset - _)
-            println("%s, %s, %s, %s, %s, %s, %s"
-              .format(group, topic, partition, offsetOpt.getOrElse("unknown"), logEndOffset, lag.getOrElse("unknown"), owner.getOrElse("none")))
-          case None => // ignore
+        if (opts.options.has(opts.newConsumerOpt)) {
+          val consumerOpt = getNewConsumer(zkUtils, brokerId)
+          consumerOpt match {
+            case Some(consumer) =>
+              consumer.assign(List(topicPartition))
+              consumer.seekToEnd(topicPartition)
+              val logEndOffset = consumer.position(topicPartition)
+              consumer.close()
+              print(logEndOffset)
+            case None => // ignore
+          }
+        } else {
+          val consumerOpt = getConsumer(zkUtils, brokerId)
+          consumerOpt match {
+            case Some(consumer) =>
+              val topicAndPartition: TopicAndPartition = new TopicAndPartition(topicPartition.topic(), topicPartition.partition())
+              val request =
+                OffsetRequest(Map(topicAndPartition -> PartitionOffsetRequestInfo(OffsetRequest.LatestTime, 1)))
+              val logEndOffset = consumer.getOffsetsBefore(request).partitionErrorAndOffsets(topicAndPartition).offsets.head
+              consumer.close()
+              print(logEndOffset)
+            case None => // ignore
+          }
         }
       case None =>
-        println("No broker for partition %s".format(topicAndPartition))
+        println("No broker for partition %s".format(topicPartition))
+    }
+  }
+
+  private def getNewConsumer(zkUtils: ZkUtils, brokerId: Int): Option[KafkaConsumer[String, String]] = {
+    try {
+      zkUtils.readDataMaybeNull(ZkUtils.BrokerIdsPath + "/" + brokerId)._1 match {
+        case Some(brokerInfoString) =>
+          Json.parseFull(brokerInfoString) match {
+            case Some(m) =>
+              val brokerInfo = m.asInstanceOf[Map[String, Any]]
+              val host = brokerInfo.get("host").get.asInstanceOf[String]
+              val port = brokerInfo.get("port").get.asInstanceOf[Int]
+              val properties: Properties = new Properties()
+              properties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, host + ":" + port)
+              properties.put(ConsumerConfig.GROUP_ID_CONFIG, "ConsumerGroupCommand")
+              properties.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "true")
```

Comment:
To be safe, we should turn off auto commit since the consumer is only used for reading partition LEO.